### PR TITLE
chores: add CI Github action

### PR DESCRIPTION
**Pull request by thedaviddias**
**Please complete these steps and check these boxes (by putting an x inside the brackets) before filing your PR:**

- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit test.

#### Short description of what this resolves:

Moving away from Travis and using Github actions.

#### Proposed changes:

- Add ci action
- Add release action using Semantic Release

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: CI integration
+name: Continuous integration
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:1]: We should use `Title Case` (`Continuous Integration`) here or just `CI`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,19 +9,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [10.x, 12.x]
+        node: [12.x]
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:12]: One build seems to take only around 20s
So we can use a `matrix.node` of 10.x (min LTS), 12.x (current LTS) and 14.x (newest) and also maybe 13.x if we want to

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,19 +9,24 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v1
+      - name: Checkout
+        uses: actions/checkout@v1
+
       - name: Use Node.js ${{ matrix.node }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node }}
           registry-url: https://registry.npmjs.org
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:22]: This may not be necessary

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -9,19 +9,24 @@ jobs:
...
+
       - name: Install dependencies
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:24]: It's helpful to add a step that prints out the exact `npm` and `node` version

File: .github/workflows/ci.yml (hunk 2 above)
Comment (thedaviddias) [reply .github/workflows/ci.yml:12]: Ok. I will add these 4 then.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (thedaviddias) [reply .github/workflows/ci.yml:22]: Which part and why?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Shinigami92) [reply .github/workflows/ci.yml:12]: Further information: https://nodejs.org/en/about/releases/

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Shinigami92) [reply .github/workflows/ci.yml:22]: `registry-url: https://registry.npmjs.org`  
I think this is used by default

File: .github/workflows/ci.yml (hunk 4 above)
Comment (thedaviddias) [reply .github/workflows/ci.yml:24]: Why would you need this to be printed as it's already returned by the job?

image

File: .github/workflows/ci.yml (hunk 2 above)
Comment (thedaviddias) [reply .github/workflows/ci.yml:12]: 13 may not be too relevant.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (Shinigami92) [reply .github/workflows/ci.yml:24]: Oh this is new to me 😮  
GitHub may add it by default now ❤

**Comment by Shinigami92** [comment]
If you want to, you can spy on my GitHub action workflow 😉 
https://github.com/prettier/plugin-pug/actions/runs/103539594/workflow

**Comment by thedaviddias** [comment]
> If you want to, you can spy on my GitHub action workflow 😉
> https://github.com/prettier/plugin-pug/actions/runs/103539594/workflow

@Shinigami92 Looking at your file, I believe I could add [push] also and not just on PR for the CI workflow, agree?

**Comment by Shinigami92** [comment]
What do you think about an `npm audit` step? (https://docs.npmjs.com/cli/audit)  
Also a `lint` step could be possible.

**Comment by thedaviddias** [comment]
> What do you think about an `npm audit` step? (https://docs.npmjs.com/cli/audit)
> Also a `lint` step could be possible.

@Shinigami92 Lint is definitely something I'm gonna put in a next PR, I'm going to add an ESLINT config I have. For the audit, I'm not sure we need this in our CI. What the differences with the github security analysis or tools like Snyk?

**Comment by Shinigami92** [comment]
> @Shinigami92 Lint is definitely something I'm gonna put in a next PR, I'm going to add an ESLINT config I have. For the audit, I'm not sure we need this in our CI. What the differences with the github security analysis or tools like Snyk?

Mh I'm not sure, I have no knowledge of Snyk
GitHub Security may only catch high warnings, but I'm not sure

On the other hand, with `audit` you have a history of actions / pipelines (don't know if this is helpful)

=> 🤷‍♂️ 😅

**Comment by thedaviddias** [comment]
@Shinigami92 Let's maybe leave this to be added in another PR :)

**Pull request merged**